src/main/java/org/codeforamerica/messaging/services/EmailService.java

```diff
@@ -41,4 +50,26 @@ private boolean unsubscribed(String toEmail) {
         EmailSubscription latestSubscription = emailSubscriptionRepository.findFirstByEmailOrderByCreationTimestampDesc(toEmail);
         return latestSubscription != null && latestSubscription.isUnsubscribed();
     }
+
+    @Transactional
+    public void updateStatus(String providerMessageId, MessageStatus newEmailStatus, String rawEmailStatus, Map<String, String> providerError) {
+        EmailMessage emailMessage = emailMessageRepository.findFirstByProviderMessageId(providerMessageId);
+        if (emailMessage == null) {
+            log.error("Cannot find message with providerId: {}", providerMessageId);
+            throw new JobRunrException("Cannot find message with providerId");
```

Comment:
Neat. This means it will be retried until the EmailMessage to update is created, or times out?